enhancements/olm/report-sdk-operators.md

```diff
@@ -0,0 +1,74 @@
+
+## Goals
+The goal of this enhancement is to list the stamps present in SDK which can be added as labels to OLM metrics.
+
+## Proposal
+
+Currently, there is a [proposal][sdk_metrics_proposal] in review which lists the bundle resources in which stamps would be added indicating the use of SDK to build an operator bundle image. 
+
+The proposed stamps have the following format in `bundle.dockerfile` and `annotations.yaml`:
+1. operators.operatorframework.io.metrics.mediatype.v1 = “metrics+v1”
+2. operators.operatorframework.io.metrics.builder = “operator-sdk-sdkversion”
+3. operators.operatorframework.io.metrics.project_layout = “sdk_project_layout/layout_version” 
+
+In CSV and CRDs, they would have the following annotations in Object Metadata:
+1. operators.operatorframework.io/builder = “operator-sdk-sdkversion”
+2. operators.operatorframework.io/project_layout = “sdk_project_layout/layout_version”
+
+At present, some of the OLM metrics track the lifecycle of CSV present in operator bundle and report its status at various stages. These stamps can be parsed by OLM from bundle resources (CSV, CRDs, `annotations.yaml` or `bundle.Dockerfile`) and added to their metrics.
+
+For example, one such metric is `csvSucceeded`. Adding of `BUILDER` and `PROJECT_LAYOUT` label to this metric will enable us to track the version and layout of operator SDK which was used to build the bundle image.
+
+The metric could be modified to:
+
+```Go
+ 	csvSucceeded = prometheus.NewGaugeVec(
+ 		prometheus.GaugeOpts{
+ 			Name: "csv_succeeded",
+ 			Help: "Successful CSV install",
+ 		},
+ 		[]string{NAMESPACE_LABEL, NAME_LABEL, VERSION_LABEL, BUILDER, PROJECT_LAYOUT},
+     )
+```
+
+Here, the value of `BUILDER` would be taken from `operators.operatorframework.io/builder` and `PROJECT_LAYOUT` would represent the value of `operators.operatorframework.io/project_layout` from CSV annotations.
+
+This data can further be added to OLM dashboard through [RedHat telemetry][cluster_monitoring] and would help us to observe the number of operators built using SDK as well as to analyze the relationship of any specific operator error with corresponding SDK version.
+
+**Note**:
+1. At this point, the two relevant metrics where these labels can be added are `csvSucceeded` and `csvAbnormal`. In future, additional labels can be added on the bundle image which would result in creation of new metrics or addition of more metric labels to existing metrics.
+2. The naming of metric labels are generic. This also enables OLM to track operators which have custom label values.
+
+## Risks and Mitigations
+Addition of labels to prometheus metrics will result in new time series and may require modification of existing PromQL queries to collect relevant data. However, this will be an one time process and hence the number of labels which are added to a metric should be restricted based on the nature of data being collected.
```

Comment:
Is there an upper bound to the number of labels?  Could a container add 1000 labels?

Reply:
There has not been any upper bound set as of it to the number of labels which would be added by OLM